src/backend/commands/extension.c

```diff
@@ -2865,9 +2885,10 @@ ApplyExtensionUpdates(Oid extensionOid,
 
 	/*
 	 * The update of extension in QE is driven by QD, so we only handle 1 version upgrade
-	 * per dispatch on the QE
+	 * per dispatch on the QE.
+	 * QE may call this function with an empty updateVersions (like at postgres) - noop
 	 */
-	AssertImply(Gp_role == GP_ROLE_EXECUTE, list_length(updateVersions) == 1);
+	AssertImply(Gp_role == GP_ROLE_EXECUTE, list_length(updateVersions) <= 1);
```

Comment:
no, if segment will call this function without command from QD (from CreateExtension) here will be a problem - updateVersions list at QE is the same as at QD, so for loop works and it leads to errors